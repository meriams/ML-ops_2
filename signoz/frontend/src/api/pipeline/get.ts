import axios from 'api';
import { ErrorResponseHandler } from 'api/ErrorResponseHandler';
import { AxiosError } from 'axios';
import { ErrorResponse, SuccessResponse } from 'types/api';
import { Pipeline } from 'types/api/pipeline/def';
import { Props } from 'types/api/pipeline/get';

const get = async (
	props: Props,
): Promise<SuccessResponse<Pipeline> | ErrorResponse> => {
	try {
		const response = await axios.get(`/logs/pipelines/${props.version}`);

		return {
			statusCode: 200,
			error: null,
			message: response.data.status,
			payload: response?.data?.data,
		};
	} catch (error) {
		return ErrorResponseHandler(error as AxiosError);
	}
};

export default get;
