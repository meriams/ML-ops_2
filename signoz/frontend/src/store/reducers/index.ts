import { combineReducers } from 'redux';

import appReducer from './app';
import globalTimeReducer from './global';
import { LogsReducer } from './logs';
import metricsReducers from './metric';
import { ServiceMapReducer } from './serviceMap';
import traceReducer from './trace';
import { usageDataReducer } from './usage';

const reducers = combineReducers({
	traces: traceReducer,
	usageDate: usageDataReducer,
	globalTime: globalTimeReducer,
	serviceMap: ServiceMapReducer,
	app: appReducer,
	metrics: metricsReducers,
	logs: LogsReducer,
});

export type AppState = ReturnType<typeof reducers>;

export default reducers;
