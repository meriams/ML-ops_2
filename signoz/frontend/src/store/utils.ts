import getLocalStorageKey from 'api/browser/localstorage/get';
import { LOCALSTORAGE } from 'constants/localStorage';
import { User } from 'types/reducer/app';

export const getInitialUserTokenRefreshToken = (): AuthTokenProps | null => {
	const accessJwt = getLocalStorageKey(LOCALSTORAGE.AUTH_TOKEN);
	const refreshJwt = getLocalStorageKey(LOCALSTORAGE.REFRESH_AUTH_TOKEN);

	if (accessJwt && refreshJwt) {
		return {
			accessJwt,
			refreshJwt,
		};
	}

	return null;
};

interface AuthTokenProps {
	accessJwt: User['accessJwt'];
	refreshJwt: User['refreshJwt'];
}
