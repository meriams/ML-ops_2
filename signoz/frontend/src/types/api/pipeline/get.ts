export type Props = {
	version: string | number;
};
