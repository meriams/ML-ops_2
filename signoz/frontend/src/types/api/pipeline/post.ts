import { PipelineData } from './def';

export interface Props {
	data: { pipelines: Array<PipelineData> };
}
