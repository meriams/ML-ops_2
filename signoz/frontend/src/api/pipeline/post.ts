import axios from 'api';
import { ErrorResponseHandler } from 'api/ErrorResponseHandler';
import { AxiosError } from 'axios';
import { ErrorResponse, SuccessResponse } from 'types/api';
import { Pipeline } from 'types/api/pipeline/def';
import { Props } from 'types/api/pipeline/post';

const post = async (
	props: Props,
): Promise<SuccessResponse<Pipeline> | ErrorResponse> => {
	try {
		const response = await axios.post('/logs/pipelines', props.data);

		return {
			statusCode: 200,
			error: null,
			message: response.data.status,
			payload: response.data.data,
		};
	} catch (error) {
		return ErrorResponseHandler(error as AxiosError);
	}
};

export default post;
