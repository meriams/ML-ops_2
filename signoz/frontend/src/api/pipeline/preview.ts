import axios from 'api';
import { ILog } from 'types/api/logs/log';
import { PipelineData } from 'types/api/pipeline/def';

export interface PipelineSimulationRequest {
	logs: ILog[];
	pipelines: PipelineData[];
}

export interface PipelineSimulationResponse {
	logs: ILog[];
}

const simulatePipelineProcessing = async (
	requestBody: PipelineSimulationRequest,
): Promise<PipelineSimulationResponse> =>
	axios
		.post('/logs/pipelines/preview', requestBody)
		.then((res) => res.data.data);

export default simulatePipelineProcessing;
