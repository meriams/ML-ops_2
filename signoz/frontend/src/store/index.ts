import {
	applyMiddleware,
	compose,
	legacy_createStore as createStore,
} from 'redux';
import thunk, { ThunkMiddleware } from 'redux-thunk';
import AppActions from 'types/actions';

import reducers, { AppState } from './reducers';

const composeEnhancers =
	(window && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

const store = createStore(
	reducers,
	composeEnhancers(
		applyMiddleware(thunk as ThunkMiddleware<AppState, AppActions>),
	),
);

export type AppDispatch = typeof store.dispatch;

if (window !== undefined) {
	window.store = store;
}

export default store;
