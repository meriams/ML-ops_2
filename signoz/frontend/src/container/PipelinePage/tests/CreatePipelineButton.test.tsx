import { render } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import i18n from 'ReactI18';
import store from 'store';

import CreatePipelineButton from '../Layouts/Pipeline/CreatePipelineButton';
import { pipelineApiResponseMockData } from '../mocks/pipeline';

describe('PipelinePage container test', () => {
	it('should render CreatePipelineButton section', () => {
		const { asFragment } = render(
			<MemoryRouter>
				<Provider store={store}>
					<I18nextProvider i18n={i18n}>
						<CreatePipelineButton
							setActionType={jest.fn()}
							isActionMode="viewing-mode"
							setActionMode={jest.fn()}
							pipelineData={pipelineApiResponseMockData}
						/>
					</I18nextProvider>
				</Provider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
