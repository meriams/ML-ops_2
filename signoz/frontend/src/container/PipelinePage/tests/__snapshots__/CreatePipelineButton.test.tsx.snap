// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipelinePage container test should render CreatePipelineButton section 1`] = `
<DocumentFragment>
  .c0.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-bottom: 2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1.c1.c1 {
  margin-left: 1rem;
}

<div
    class="c0"
  >
    <span
      aria-label="question-circle"
      class="anticon anticon-question-circle"
      role="img"
      style="font-size: 1rem; color: rgba(255, 255, 255, 0.835);"
    >
      <svg
        aria-hidden="true"
        data-icon="question-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 708c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm62.9-219.5a48.3 48.3 0 00-30.9 44.8V620c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-21.5c0-23.1 6.7-45.9 19.9-64.9 12.9-18.6 30.9-32.8 52.1-40.9 34-13.1 56-41.6 56-72.7 0-44.1-43.1-80-96-80s-96 35.9-96 80v7.6c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V420c0-39.3 17.2-76 48.4-103.3C430.4 290.4 470 276 512 276s81.6 14.5 111.6 40.7C654.8 344 672 380.7 672 420c0 57.8-38.1 109.8-97.1 132.5z"
        />
      </svg>
    </span>
    <button
      class="ant-btn css-dev-only-do-not-override-2i2tap ant-btn-default c1"
      type="button"
    >
      <span
        class="ant-btn-icon"
      >
        <span
          aria-label="edit"
          class="anticon anticon-edit"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="edit"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32zm-622.3-84c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9z"
            />
          </svg>
        </span>
      </span>
      <span>
        enter_edit_mode
      </span>
    </button>
  </div>
</DocumentFragment>
`;
